internal/mesh/internal/controllers/sidecarproxy/fetcher/data_fetcher.go

```diff
@@ -96,6 +112,30 @@ func (f *Fetcher) FetchServiceEndpoints(ctx context.Context, id *pbresource.ID)
 	return se, nil
 }
 
+func (f *Fetcher) FetchService(ctx context.Context, id *pbresource.ID) (*intermediateTypes.Service, error) {
+	rsp, err := f.Client.Read(ctx, &pbresource.ReadRequest{Id: id})
+
+	switch {
```

Comment:
[nit] would this logic be better in the consumer of the function (`FetchImplicitDestinationsData()`) since we already do another `if not found vs if err` check there and we are effectively doing this twice?

Reply:
I think I'll need to change a bunch of these in a future PR to use RB's decoded resource functions and it'll probably get rid of or change most of this logic. I'll leave it as is for now with that thought in mind.